### ISSUE TITLE
Invalid JS: Compiler emits invalid JS when `throw` followed by `if` or `for`

### ISSUE THREAD

**Issue by epidemian** [jashkenas/coffeescript#3709]
This seemingly valid CS code:

``` coffeescript
throw if pedantic 
  new FatalError
else
  new ResumableError
```

Compiles to invalid JS:

``` javascript
throw if (pedantic) {
  new FatalError;
} else {
  new ResumableError;
};
```

I say seemingly valid because the same thing works for a function call:

``` coffeescript
raise = (e) -> throw e
raise if pedantic 
  new FatalError
else
  new ResumableError
```

``` javascript
var raise;

raise = function(e) {
  throw e;
};

raise(pedantic ? new FatalError : new ResumableError);
```

**Comment by GeoffreyBooth** [comment]
Probably related, from #3357: The following [compiles to invalid JS](http://localhost/v2/#try:throw%20for%20v%20in%20list%20then%20v):

```coffee
throw for v in list then v
```

Basically we shouldn’t allow `if` or `for` (or all expressions?) after `throw`.

**Comment by lydell** [comment]
@GeoffreyBooth You seem to have linked to localhost.

**Comment by GeoffreyBooth** [comment]
Fixed by #4664.
